### PR TITLE
`General`: Fix team and start exercise button visible if student is not assigned to a team

### PR DESCRIPTION
**Pull request by tobias-lippert**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

In the current version, the team button and the the start exercise are visible and clickable even if the student is not assigned to a team yet. This leads to a bad request/internal server error if the buttons are clicked
This PR fixes the condition when these buttons are shown. 

### Description
Fixed the condition for the team button (it was incorrectly negated)
added conditions to hide the start exercise button for team exercises if the student is not yet part of a team.
### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students 

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a team exercise that is released (affected exercise types are file-upload, modeling and text)
4. navigate as a student to the exercise and verify no buttons are visible
5. add the student to a team
6. Navigate again as a student to the exercise and verify that both buttons are visible, clickable and no errors are produced when clicking them.

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

my changes are covered
### Screenshots

Before
image

After
image

### CODE REVIEW

File: src/test/javascript/spec/component/overview/exercise-details/exercise-details-student-actions.component.spec.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ import { MockRouter } from '../../../helpers/mocks/mock-router';
 import { MockCourseExerciseService } from '../../../helpers/mocks/service/mock-course-exercise.service';
 import { MockSyncStorage } from '../../../helpers/mocks/service/mock-sync-storage.service';
 import { ArtemisTestModule } from '../../../test.module';
+import { ModelingExercise } from 'app/entities/modeling-exercise.model';
```
Comment (b-fein) [inline src/test/javascript/spec/component/overview/exercise-details/exercise-details-student-actions.component.spec.ts:36]:
```suggestion
```
Unused import

File: src/test/javascript/spec/component/overview/exercise-details/exercise-details-student-actions.component.spec.ts [hunk 2]
```diff
@@ -45,15 +46,15 @@ describe('ExerciseDetailsStudentActionsComponent', () => {
     let getProfileInfoSub: jest.SpyInstance;
 
     const team = { id: 1, students: [{ id: 99 } as User] } as Team;
-    const programmingExercise: ProgrammingExercise = {
+    const programmingExercise: Exercise = {
```
Comment (b-fein) [inline src/test/javascript/spec/component/overview/exercise-details/exercise-details-student-actions.component.spec.ts:49]: Variable name should be updated as well.

**Review by jakubriegel — DISMISSED** [review]
Tested locally, works fine 👍🏻

File: src/test/javascript/spec/component/overview/exercise-details/exercise-details-student-actions.component.spec.ts [hunk 3]
```diff
@@ -102,38 +102,46 @@ describe('ExerciseDetailsStudentActionsComponent', () => {
...
-        expect(viewTeamButton).not.toBeNull();
-    }));
+            const viewTeamButton = fixture.debugElement.query(By.css('.view-team'));
+            expect(viewTeamButton).not.toBeNull();
+        }),
+    );
 
-    it('should show the button "Start exercise" for a team exercise when assigned to a team', fakeAsync(() => {
-        comp.exercise = teamExerciseWithTeamAssigned;
+    it.each([ExerciseType.PROGRAMMING, ExerciseType.TEXT, ExerciseType.FILE_UPLOAD, ExerciseType.MODELING])(
```
Comment (julian-christl) [inline src/test/javascript/spec/component/overview/exercise-details/exercise-details-student-actions.component.spec.ts:133]: Shouldn't we be able to combine this test with the one above as we did with the other two tests?

**Review by julian-christl — COMMENTED** [review]
Simple fix, looking good codewise. I think we can merge two tests, though, what do you think?

**Review by julian-christl — APPROVED** [review]
Thanks for the update, code approved

**Review by b-fein — APPROVED** [review]
Code :+1:

**Review by JohannesStoehr — APPROVED** [review]
Code changes and new tests look good to me

**Review by terlan98 — APPROVED** [review]
Tested on TS2. Works as expected 👍  The buttons now appear only when the user is a part of a team.

**Review by jakubriegel — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Pull request merged**